src/openms/source/KERNEL/MSExperiment.cpp

```diff
@@ -337,6 +329,12 @@ namespace OpenMS
   /// returns an array of MS levels
   const std::vector<UInt>& MSExperiment::getMSLevels() const
   {
+    // if we have spectra enforce MS level information
+    // prevents uninitialized use uf ms_levels_
+    if (!spectra_.empty() && ms_levels_.empty()) 
+    {
+      throw Exception::InvalidSize(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, spectra_.size());
```

Comment:
can you use another exception with a better error message?
If a user sees what is currently thrown, its not very helpful, since 'InvalidSize' does not allow for a message (or maybe add one, that's also fine).